### ISSUE TITLE
loss function input

### ISSUE THREAD

**Issue by wuqun-tju** [crockwell/far#4]
hello, 

in the code [here](https://github.com/crockwell/far/blob/main/mapfree_6dreg/lib/models/regression/model.py#L312), rot_loss's input (data['R']) is 6 x 1, but [here](https://github.com/crockwell/far/blob/main/mapfree_6dreg/lib/utils/loss.py#L88) , the Input R is 3x3, it looks like they are not the same dim. I appreciate your response！

**Comment by crockwell** [comment]
The code runs, no? I think it is transformed from rotation matrix to 6D coords and back at some point. Checkout https://arxiv.org/abs/1812.07035 if you're curious about the differences.

**Comment by wuqun-tju** [comment]
Thank you , I found it is set 'rot_6d_loss' in config rot6d_trans_with_loftr.yaml , not rot_frobenius_loss in default.yaml

**Comment by wuqun-tju** [comment]
I read the paper you post, in the paper , the R‘s 3rd colum can be dropped 
![image](https://github.com/crockwell/far/assets/58416153/b1ab2128-dda2-42e5-a918-fa8de228e40b)
but in your code, R's 3rd row is dropped. the reason maybe is R.T = R ? could you explain this difference for me,  Thanks!

**Comment by wuqun-tju** [comment]
Sorry for another question, the rot_6D_loss function just compute 6D vector l1_loss between Rgt and R_6d，does it make sense? could you help understand it , Thank you very much

**Comment by crockwell** [comment]
Agreed -- the 6D loss and converting to a rotation matrix from only the first two rows (hence the "6D") is the recipe prescribed in https://arxiv.org/abs/1812.07035

To summarize, a 6D representation is continuous and can be mapped to a rotation matrix, which makes optimization easier, compared to training directly on nonlinear output of e.g. quaternions. If you're curious for a more in-depth analysis of their method, I'd recommend reaching out to the authors of the original work.
